kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -89,13 +92,26 @@ public CloseableIterator<FilteredColumnarBatch> getScanFiles(TableClient tableCl
             throw new IllegalStateException("Scan files are already fetched from this instance");
         }
         accessedScanFiles = true;
+
+        // Generate data skipping filter and decide if we should read the stats column
+        Optional<Predicate> dataFilter = getDataSkippingPredicate();
```

Comment:
dataFilter -> dataSkippingFilter

also... why alternate between "filter" and "predicate"?